airflow/contrib/hooks/gcs_hook.py

```diff
@@ -265,6 +275,10 @@ def delete(self, bucket, object):
         :param object: name of the object to delete
         :type object: str
         """
+
+        if generation is not None:
+            warnings.warn("'generation' parameter is no longer supported", DeprecationWarning)
```

Comment:
We should add a `stacklevel=2` to all these warn calls - that way the warning will be reported with the filename/line number of the caller, rather than `gcs_hook;py:280: generation' parameter is no longer supported`